File: ffi/capi/bindings/js/LineBreakStrictness.mjs
import wasm from "./diplomat-wasm.mjs"
import * as diplomatRuntime from "./diplomat-runtime.mjs"

export const LineBreakStrictness_js_to_rust = {
  "Loose": 0,
  "Normal": 1,
  "Strict": 2,
  "Anywhere": 3,
};

export const LineBreakStrictness_rust_to_js = {
  [0]: "Loose",
  [1]: "Normal",
  [2]: "Strict",
  [3]: "Anywhere",
};

export const LineBreakStrictness = {
  "Loose": "Loose",
  "Normal": "Normal",
  "Strict": "Strict",
  "Anywhere": "Anywhere",
};

File: ffi/capi/bindings/kotlin/src/main/kotlin/org/unicode/icu4x/LineBreakStrictness.kt
package org.unicode.icu4x

import com.sun.jna.Callback
import com.sun.jna.Library
import com.sun.jna.Native
import com.sun.jna.Pointer
import com.sun.jna.Structure

internal interface LineBreakStrictnessLib: Library {
}
/** See the [Rust documentation for `LineBreakStrictness`](https://docs.rs/icu/2.1.1/icu/segmenter/options/enum.LineBreakStrictness.html) for more information.
*/
enum class LineBreakStrictness {
    Loose,
    Normal,
    Strict,
    Anywhere;

    fun toNative(): Int {
        return this.ordinal
    }


    companion object {
        internal val libClass: Class<LineBreakStrictnessLib> = LineBreakStrictnessLib::class.java
        internal val lib: LineBreakStrictnessLib = Native.load("icu4x", libClass) 
        fun fromNative(native: Int): LineBreakStrictness {
            return LineBreakStrictness.entries[native]
        }

        fun default(): LineBreakStrictness {
            return Loose
        }
    }
}